notification-service/src/main/java/com/redhat/parodos/notification/exceptions/StateNotFoundOrUnsupportedException.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022 Red Hat Developer
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.parodos.notification.exceptions;
+
+/**
+ * The StateNotFoundOrUnsupportedException wraps unchecked standard Java exception and
+ * enriches them with a custom error code. You can use this execution when a Notification
+ * State is not found or is unsupported.
+ *
+ * @author Gloria Ciavarrini (Github: gciavarrini)
+ */
+public class StateNotFoundOrUnsupportedException extends RuntimeException {
+
+	private static final long serialVersionUID = 1L;
+
+	public StateNotFoundOrUnsupportedException() {
+	}
+
+	public StateNotFoundOrUnsupportedException(String message) {
+		super(message);
+	}
+
+	public StateNotFoundOrUnsupportedException(String message, Throwable cause) {
+		super(message, cause);
```

Comment:
sonarqube analysis will fail if method is not referenced, can we remove this for now and add back when needed

Reply:
Removed all unused methods from all the custom exceptions